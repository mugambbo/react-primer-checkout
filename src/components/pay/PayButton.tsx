import React from 'react'
import I18nHelper from '../../helpers/I18nHelper';
import { useI18nContext } from '../../i18n/I18nContext'
import { PayButtonStyles } from '../checkout/CheckoutContainer'


interface PayButtonProps {
    btnStyles?: PayButtonStyles;
    loading: boolean;
    amount?: number;
    currency?: string;
}

export enum BtnVariant {
  small,
  large
}

function PayButton({ btnStyles, amount, currency, loading }: PayButtonProps) {

  const contentString = useI18nContext();

    return (
        <button id="pay" type="submit" className={"pay-btn" + (btnStyles?.btnVariant === BtnVariant.small? " pay-btn-small": "") + (loading? " pay-btn-loading": "")} style={{...btnStyles?.style}} disabled={loading} >
          {loading? (
          <div className="checkout-center">
            <div className="dot-windmill" role="progressbar" aria-valuetext={I18nHelper.formatString(contentString.aria_pay_status)} ></div>
          </div>
          ): (
            <div className="checkout-center">
              <img style={{marginRight: 8}} src={btnStyles?.logoSrc} height="24px" alt="" />{I18nHelper.formatString(contentString.pay, {0: currency, 1: amount})}
            </div>
          )}
        </button>
    )
}

export default PayButton
