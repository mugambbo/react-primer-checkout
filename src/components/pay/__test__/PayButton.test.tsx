import React from 'react';
import { shallow, mount } from 'enzyme';
import PayButton from '../PayButton';

describe('<PayButton />', () => {
    test('renders the pay button: default', () => {
        const loading = false;
        const component = shallow(<PayButton loading={loading} />);  
        expect(component).toMatchSnapshot();
    });

    test('renders the pay button: loading', () => {
        const loading = true;
        const component = shallow(<PayButton loading={loading} />);  
        expect(component).toMatchSnapshot();
    });

    test('Displays the pay button correctly', () => {
        const loading = false;
        const amount = 50;
        const currency = "$";
        const wrapper = shallow(<PayButton amount={amount} currency={currency} loading={loading} />);  
        const text = wrapper.find("button").first().text();
        expect(text).toEqual(`PAY ${currency}${amount}`);
    });

    test('Displays the image beside pay button correctly', () => {
        const loading = false;
        const expectedImageSrc = "https://react-checkout-primer.com/logo.png";
        const wrapper = shallow(<PayButton btnStyles={{logoSrc: expectedImageSrc}} loading={loading} />);  
        const actualImageSrc = wrapper.find("img").first().prop("src");
        expect(actualImageSrc).toEqual(expectedImageSrc);
    });
});