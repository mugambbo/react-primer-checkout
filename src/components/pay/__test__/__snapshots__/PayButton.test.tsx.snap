// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PayButton /> renders the pay button: default 1`] = `
<button
  className="pay-btn"
  disabled={false}
  id="pay"
  style={Object {}}
  type="submit"
>
  <div
    className="checkout-center"
  >
    <img
      alt=""
      height="24px"
      style={
        Object {
          "marginRight": 8,
        }
      }
    />
    PAY {0}{1}
  </div>
</button>
`;

exports[`<PayButton /> renders the pay button: loading 1`] = `
<button
  className="pay-btn pay-btn-loading"
  disabled={true}
  id="pay"
  style={Object {}}
  type="submit"
>
  <div
    className="checkout-center"
  >
    <div
      aria-valuetext="Connecting to payment gateway, please wait..."
      className="dot-windmill"
      role="progressbar"
    />
  </div>
</button>
`;
